### ISSUE TITLE
Add aria-label to month grid

### ISSUE THREAD

**Issue by gpbl** [gpbl/react-day-picker#1350]
According to [WAI-aria-practices](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html), the table grid for the month is missing the caption element.

A solution is to add [@reach/auto-id](https://reach.tech/auto-id/) package to give an `id` to the caption heading element. The id should be passed to the table's `aria-labelled-by`. Special case when the caption is in dropdown mode, where the heading element may be missing.
